fibonacci.js

```diff
@@ -0,0 +1,110 @@
+let getNumber = document.getElementById("numberHere");
+let fibURL = "http://localhost:5050/fibonacci/";
+let answer = document.getElementById("answer");
+let mySpinner = document.getElementById("spinner");
+let button = document.getElementById("submitButton");
+let li = document.getElementsByClassName("li");
+let alertBox = document.getElementById("alert");
+const resultsListurl = "http://localhost:5050/getFibonacciResults";
+const resultsListElement = document.getElementById("results-list");
+
+function getFib() {
+  mySpinner.classList.remove("visually-hidden");
+  answer.classList.add("visually-hidden");
+  if (getNumber.value > 50) {
+    alertBox.classList.remove("visually-hidden");
+    spinner.classList.add("visually-hidden");
+  }
+  if (getNumber.value < 50) {
+    fetch(`${fibURL}${getNumber.value}`)
+      .then(function (response) {
+        if (!response.ok) {
+          response.text().then((error) => {
+            answer.classList.remove(
+              "visually-hidden",
+              "text-decoration-underline",
+              "fw-bold"
+            );
+            answer.classList.add("text-danger");
+            alertBox.classList.add("visually-hidden");
+            getNumber.classList.add("border-danger");
+            answer.innerText = error;
+            mySpinner.classList.toggle("visually-hidden");
+          });
+        } else return response.json();
+      })
+
+      .then((data) => {
+        answer.innerHTML = data.result;
+      })
+
+      .then((spinner) => {
+        mySpinner.classList.toggle("visually-hidden");
+        answer.classList.remove("visually-hidden", "text-danger");
+        answer.classList.add("text-decoration-underline", "fw-bold");
+        getNumber.classList.remove("border-danger");
+        alertBox.classList.add("visually-hidden");
+      });
+  }
+}
+
+function getResults() {
+  fetch(resultsListurl)
+    .then((response) => response.json())
+    .then((data) => {
+      printResultstoList(data);
+    })
+    .then(fetch(resultsListurl));
+}
+
+function printResultstoList(data) {
+  let myData = data.results;
+  myData.sort((a, b) => b.createdDate - a.createdDate);
+  resultsListElement.innerHTML = "";
+  for (i = 0; i < 10; i++) {
```

Comment:
why should the loop run only 10 times? what if your database (history) has more then 10 results?